### DISCUSSION TITLE
Bug? isValidating is false for stale data

### DISCUSSION THREAD

**Discussion by fabb** [vercel/swr#1271] [Q&A]
I have a very specific use case where I have not yet found a solution with useSWR. Basically I have a paged search result, and when the user tries to open a page that does not exist, the data should be resetted to the last existing page. I've implemented this similar to this simplified code:

```
interface SearchResult {
    elements: unknown[]
    pages: number
    // count of elements on ALL pages of the search result
    count: number
}

...

const [filters, setFilters] = useState([page, sortOrder, searchText])
const { data, error, mutate, isValidating } = useSWR<SearchResult, Error>(['search', ...filters])

// when a page after the last page has been requested, reset to the last page
if (!isValidating && data && data?.count > 0 && data?.elements.length === 0) {
    setFilters(([page, ...remainingFilters]) => [data.pages, ...remainingFilters])
}
```

This mostly works except for one case:
1. I start with 51 elements while the page size is 50
2. I delete the last element
3. Automatically by the code in the last `if`, I get thrown back to page 1 because this is the last page
4. I create a new element in its own flow, but I don't `mutate` my search result since I would need to do it for all permutations of the filters which is impossible - at the end of the flow I end up on page 1 where the new element shows up
5. I try to go to page 2

Now the problematic part happens. Page 2 returns the stale response from after step 2 which has empty `elements`. This means the code in the `if` kicks in and throws me back to page 1, even before the response can be revalidated. **Curiously `isValidating` is `false` - is this a bug?**
So to sum up, when I click the `page 2` button, nothing happens for the user, only the second click gets the user to page 2 because the first attempt revalidated the second page.

Ok, so 2 questions:
1. Is this a bug with `isValidating`?
2. Are there other suggestions on how to solve my issue? Note that I want to avoid to preload pages because of DB load concerns.
